src/Analysis/Ast/Impl/Documents/RunningDocumentTable.cs

```diff
@@ -223,7 +224,8 @@ private class DocumentEntry {
 
         public void Dispose() {
             lock (_lock) {
-                foreach (var d in _documentsByUri.Values.OfType<IDisposable>()) {
+                // DocumentEntry is not disposable. it is Document that is disposable.
+                foreach (var d in _documentsByUri.Values.Select(v => v.Document)) {
```

Comment:
existing code didn't do anything since Value is DocumentEntry and it doesn't implement Disposable. I can either remove code here or make it to dispose Document.

Reply:
It selects document from the value, no?

Reply:
now d is IDocument. before it was DocumentEntry which doesn't implement IDisposable